### PR TITLE
Release 1.4.0

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

**Review by petervdm — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- Added a workaround for an NHAIS issue where the DTM segment contains a timestamp is the past. The adaptor now reads its
+ translationTimestamp from the interchange header.
+- Encapsulate outboundState recep fields in sub-document. Updated the documentation for reporting on missing RECEPs. No
+ data migration in place; to generate reports on pre-1.4.0 records follow the instructions from version 1.3.1.
+- The inboundState and outboundState.recep documents now contain a 'processedTimestamp' as a record of when the adaptor 
+ received and processed the transaction.
+- Adds a ConversationId header to outbound API requests, inbound GP system message queue, related internal queues, 
+ and database documents. Logs related to processing that transaction include the ConversationId value.
+- Replaced an embedded trust store with the capability to download a trust store (Java Keystore) at start-up when
```
Comment (bartek-sarul) [inline CHANGELOG.md:19]: I would add info here that the custom trust store is combined with the standard java one. Not replacing it

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/configuration/AppConfiguration.java [hunk 2]
```diff
@@ -21,7 +21,10 @@
 
     @Bean
     public AmazonS3 getS3Client() {
-        return AmazonS3ClientBuilder.standard()
-            .build();
+        if(trustStoreUrl != null && trustStoreUrl.startsWith("s3")) {
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/configuration/AppConfiguration.java:24]: Why this change? Client initialization fails if AWS is not available or is this just an optimization step?

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/configuration/AppConfiguration.java (hunk 2 above)
Comment (mattd-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/configuration/AppConfiguration.java:24]: App doesn't start if there are no AWS environment variables on the PATH. In this case, it is the region that is required.

File: CHANGELOG.md (hunk 1 above)
Comment (mattd-kainos) [reply CHANGELOG.md:19]: 👍 added a sentence to the README to clarify that the additional certificates are added to the default trust store.

**Review by bartek-sarul — APPROVED** [review]

**Pull request merged**